From 89e70643d72974b610f355701a767ea0d0538104 Mon Sep 17 00:00:00 2001
From: "Daniel P. Berrange" <berrange@redhat.com>
Date: Fri, 12 Oct 2012 11:44:18 +0100
Subject: [PATCH] Do not set VIR_TYPED_PARAM_STRING_OKAY flag

The VIR_TYPED_PARAM_STRING_OKAY flag is set automatically by
libvirt when required
(cherry picked from commit 04a7eae3602c0097d89a8b371dc8cd13776a07e7)
rhbz #848309
---
 Virt.xs | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/Virt.xs b/Virt.xs
index 6bd3448..2f22909 100644
--- a/Virt.xs
+++ b/Virt.xs
@@ -1248,13 +1248,12 @@ vir_typed_param_to_hv(virTypedParameter *params, int nparams)
 }
 
 
-static int
+static void
 vir_typed_param_from_hv(HV *newparams, virTypedParameter *params, int nparams)
 {
     unsigned int i;
     char * ptr;
     STRLEN len;
-    int needString = 0;
 
     for (i = 0 ; i < nparams ; i++) {
         SV **val;
@@ -1290,13 +1289,11 @@ vir_typed_param_from_hv(HV *newparams, virTypedParameter *params, int nparams)
             break;
 
         case VIR_TYPED_PARAM_STRING:
-            needString = 1;
             ptr = SvPV(*val, len);
             params[i].value.s = (char *)ptr;
             break;
         }
     }
-    return needString;
 }
 
 
@@ -3288,7 +3285,6 @@ set_blkio_parameters(dom, newparams, flags=0)
   PREINIT:
       virTypedParameter *params;
       int nparams;
-      int needString;
     PPCODE:
       nparams = 0;
       if (virDomainGetBlkioParameters(dom, NULL, &nparams, flags) < 0)
@@ -3301,10 +3297,10 @@ set_blkio_parameters(dom, newparams, flags=0)
           _croak_error();
       }
 
-      needString = vir_typed_param_from_hv(newparams, params, nparams);
+      vir_typed_param_from_hv(newparams, params, nparams);
 
       if (virDomainSetBlkioParameters(dom, params, nparams,
-                                      needString ? flags | VIR_TYPED_PARAM_STRING_OKAY: flags) < 0)
+                                      flags) < 0)
           _croak_error();
       Safefree(params);
 
